RecursiveExtractor/Extractors/TarExtractor.cs

```diff
@@ -108,14 +115,16 @@ public IEnumerable<FileEntry> Extract(FileEntry fileEntry, ExtractorOptions opti
             }
             else
             {
-                foreach (TarArchiveEntry tarEntry in archive.Entries)
+                TarEntryCollectionEnumerator tarEntryCollectionEnumerator = new(archive.Entries, fileEntry.FullPath);
+                while(tarEntryCollectionEnumerator.GetNextEntry() != null)
                 {
+                    var tarEntry = tarEntryCollectionEnumerator.CurrentEntry;
                     if (tarEntry.IsDirectory)
                     {
                         continue;
                     }
 
-                    var fs = new FileStream(TempPath.GetTempFilePath(), FileMode.Create, FileAccess.ReadWrite, FileShare.ReadWrite, bufferSize, FileOptions.DeleteOnClose);
+                    var fs = StreamFactory.GenerateAppropriateBackingStream(options, tarEntry.Size);
```

Comment:
I have the same concern around a missing `using`/`Dispose` as line 57.

Reply:
Opened this issue to track harmonizing the design patterns here: https://github.com/microsoft/RecursiveExtractor/issues/154.